### PR TITLE
integrate udp rendezvous server

### PR DESCRIPTION
**Pull request by canndrew**

### CODE REVIEW

File: src/net/listener.rs [hunk 1]
```diff
@@ -321,6 +322,7 @@ mod test {
             })
             .join({
                 socket_incoming
+                .map_err(|e| panic!("incoming error: {}", e))
```
Comment (povilasb) [inline src/net/listener.rs]: I guess you accidentally left `panic!`?

File: src/net/protocol_agnostic/listener.rs [hunk 2]
```diff
@@ -140,37 +191,132 @@ impl PaListener {
...
                 }
+                Ok(Async::NotReady)
             }
-            PaIncoming::Utp(ref mut incoming) => {
-                match incoming.poll()? {
+            PaIncomingInner::Utp { ref mut incoming, ref mut raw_rx, ref mut processing } => {
+                loop {
+                    match raw_rx.poll().void_unwrap() {
+                        Async::Ready(Some(raw_channel)) => {
+                            processing.push({
```
Comment (povilasb) [inline src/net/protocol_agnostic/listener.rs]: This body of this match arm could be extracted to separate function. That would make code a little more readable, maintainable, testable, .etc

File: src/net/protocol_agnostic/listener.rs [hunk 3]
```diff
@@ -140,37 +191,132 @@ impl PaListener {
...
-                        let addr = PaAddr::Tcp(addr);
-                        Ok(Async::Ready(Some((stream, addr))))
+    fn poll(&mut self) -> Result<Async<Option<(PaStream, PaAddr)>>, AcceptError> {
+        match self.inner {
+            PaIncomingInner::Tcp { ref mut incoming, ref mut processing } => {
+                loop {
+                    match incoming.poll().map_err(AcceptError::TcpAccept)? {
+                        Async::Ready(Some((stream, addr))) => {
+                            processing.push({
+                                let header = [0u8; 8];
```
Comment (povilasb) [inline src/net/protocol_agnostic/listener.rs]: This body of this match arm could be extracted to separate function. That would make code a little more readable, maintainable, testable, .etc

**Comment by povilasb** [comment]
Also, `rustfmt` is failing.

File: src/net/listener.rs (hunk 1 above)
Comment (canndrew) [reply src/net/listener.rs]: Nah, it's in a test, it's supposed to panic.

Comment (povilasb) [reply src/net/listener.rs]: right, didn't check context :)

File: src/net/listener.rs.bk [hunk 4]
```diff
@@ -0,0 +1,420 @@
+// Copyright 2017 MaidSafe.net limited.
```
Comment (povilasb) [inline src/net/listener.rs.bk]: Woops, accidentally comited file?

Comment (canndrew) [reply src/net/listener.rs.bk]: Yep, fixed.

**Comment by povilasb** [comment]
Are you planning to squash the commits? Because commit messages don't satisfy our conventions.

**Comment by canndrew** [comment]
Yeah, I also need to make it point to real versions of `p2p` and `tokio-utp` though.

**Comment by canndrew** [comment]
I've published tokio-utp-0.3.0. We're also waiting on https://github.com/ustulation/p2p/pull/33 to be merged before this can be unblocked.

**Pull request merged**